addict/addict.py

```diff
@@ -76,6 +76,15 @@ def __setitem__(self, name, value):
         """
         super(Dict, self).__setitem__(name, value)
 
+    def __add__(self, other):
+        if len(self.keys()) == 0:
+            return other
+        else:
+            self_type = type(self).__name__
+            other_type = type(other).__name__
+            msg = "unsupported operand type(s) for +: '{}' and '{}'"
```

Comment:
I wanted it to follow the same format as the base TypeError for +:

```
>>> 1 + 'str'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'int' and 'str'
```

```
>>> from addict import Dict
>>> d = Dict()
>>> d.x.y = 'defined'
>>> d.x += 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "addict/addict.py", line 86, in __add__
    raise TypeError(msg.format(self_type, other_type))
TypeError: unsupported operand type(s) for +: 'Dict' and 'int'
```

Let me know if you still want to change it.